Test/HigherOrderIntrinsicSpecification/ReadPreconditionBypass3.dfy

```diff
@@ -0,0 +1,32 @@
+// RUN: %exits-with 4 %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+class Ref {
+  var inner: Ref
+  constructor()
+}
+
+function myf(o: Ref): ()
+  requires false
+  reads o.inner
+{
+  ()
+}
+
+method Main()
+  ensures false
+{
+  var outer := new Ref();
+
+  var myg := myf.requires;
+
+  var inner1 := new Ref();
+  outer.inner := inner1;
+  var reads1 := myg.reads(outer);
+  assert reads1 == {inner1}; // Error: assertion might not hold
```

Comment:
Because `f.requires.reads(t) == f.reads(t)` is replaced by `f.requires(t) ==> f.requires.reads(t) == f.reads(t)` and `myf.requires(outer) == false`?